### PR TITLE
add cloud kms

### PR DESCRIPTION
**Pull request by HaoXuan40404**

### CODE REVIEW

File: bcos-security/bcos-security/cloudkms/AwsKmsWrapper.cpp [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+#include "Common.h"
+#include "bcos-utilities/FileUtility.h"
+#include <aws/core/auth/AWSCredentials.h>
+#include <aws/core/utils/Array.h>
+#include <aws/kms/model/DecryptRequest.h>
+#include <aws/kms/model/EncryptRequest.h>
+namespace bcos::security
+{
+
+AWSKMSWrapper::AWSKMSWrapper(const std::string& region, const std::string& accessKey,
```
Comment (bxq2011hust) [inline bcos-security/bcos-security/cloudkms/AwsKmsWrapper.cpp:33]:
```suggestion
AwsKmsWrapper::AWSKMSWrapper(const std::string& region, const std::string& accessKey,
```

File: bcos-security/bcos-security/cloudkms/CloudKmsInterface.cpp [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+                        << LOG_KV("Invalid KMS provider:", m_kmsType)
+                        << LOG_KV("KMS url:", m_kmsUrl);
+        BOOST_THROW_EXCEPTION(KmsTypeError());
+    }
+
+    if (provider == CloudKmsProvider::AWS)
+    {
+        // initialize the AWS SDK
+        std::vector<std::string> awsKmsUrlParts;
+        std::string _region;
```
Comment (bxq2011hust) [inline bcos-security/bcos-security/cloudkms/CloudKmsInterface.cpp:59]:
```suggestion
        std::string region;
```

File: bcos-security/bcos-security/cloudkms/CloudKmsInterface.h [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+    std::shared_ptr<bytes> decryptContents(const std::shared_ptr<bytes>& contents) override;
+    std::shared_ptr<bytes> decryptFile(const std::string& filename) override;
+
+protected:
+    std::string m_kmsType;
+    std::string m_kmsUrl;
+
+private:
+    // for kms
+    static bool splitKmsUrl(const std::string& _kmsUrl, std::vector<std::string>& result);
```
Comment (bxq2011hust) [inline bcos-security/bcos-security/cloudkms/CloudKmsInterface.h:50]: don't use static member

**Review by bxq2011hust — APPROVED** [review]

File: bcos-security/bcos-security/cloudkms/CloudKmsInterface.cpp [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+        std::string secretKey = awsKmsUrlParts[2];
+
+        std::shared_ptr<bytes> decryptResult = nullptr;
+        Aws::SDKOptions options;
+        Aws::InitAPI(options);
+        {
+            try
+            {
+                // create an AWS KMS wrapper
+                AwsKmsWrapper kmsWrapper(region, accessKey, secretKey);
```
Comment (JimmyShi22) [inline bcos-security/bcos-security/cloudkms/CloudKmsInterface.cpp:77]: Why not create kmsWrapper beforehand as an class member?

**Review by JimmyShi22 — APPROVED** [review]

**Pull request merged**